lib/stripe/api_operations/nested_resource.rb

```diff
@@ -53,12 +54,30 @@ def nested_resource_class_methods(resource, path: nil, operations: nil,
             )
           end
         when :retrieve
+          # TODO: (Major) Split params_or_opts to params and opts and get rid of the complicated way to add params
           define_singleton_method(:"retrieve_#{resource}") \
-              do |id, nested_id, opts = {}|
+              do |id, nested_id, params_or_opts = {}, definitely_opts = nil|
+            opts = nil
+            params = nil
+            if definitely_opts.nil?
+              unrecognized_key = params_or_opts.keys.find { |k| !Util::OPTS_USER_SPECIFIED.include?(k) }
+              if unrecognized_key
+                raise ArgumentError,
+                      "Unrecognized request option: #{unrecognized_key}. Did you mean to specify this as " \
+                      "retrieve params? " \
+                      "If so, you must explicitly pass an opts hash as a third argument. " \
+                      "For example: .retrieve({#{unrecognized_key}: 'foo'}, {})"
```

Comment:
I believe the error should be something like this (because of `id` + `nested_id`)
```suggestion
                      "Unrecognized request option: #{unrecognized_key}. Did you mean to specify this as " \
                      "retrieve params? " \
                      "If so, you must explicitly pass an opts hash as a fourth argument. " \
                      "For example: .retrieve(#{id}, #{nested_id}, {#{unrecognized_key}: 'foo'}, {})"
```